File: AndroidN_Jack_Java8/app/src/main/java/ca/six/java8/MainActivity.java
package ca.six.java8;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import rx.Observable;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btn = (Button) findViewById(R.id.button2);
        btn.setOnClickListener(v -> {
            System.out.println("szw");

            Observable.just(23, 24)
                    .map(it -> "test "+it)
                    .subscribe(it -> System.out.println("Result = "+it));
        });
    }
}

File: Thread_I54W/app/src/main/java/ca/six/thread_i54w/MainActivity.kt
package ca.six.thread_i54w

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import java.util.concurrent.CountDownLatch
import java.util.concurrent.CyclicBarrier
import kotlin.random.Random

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnMain.setOnClickListener {
        }
    }

}


